### ISSUE TITLE
click is working a little differently than native-dom-helpers

### ISSUE THREAD

**Issue by kellyselden** [emberjs/ember-test-helpers#365]
The following code works as expected with ember-native-dom-helpers:

```js
    let promise = click('[data-test-confirm-delete-modal-button]');

    assert.ok(find('[data-test-deleting-modal]'));

    deferred.resolve();

    await promise;

    assert.notOk(find('[data-test-deleting-modal]'));
```

When switching over to the new helpers, `click` never actually clicks anything until the `await promise;` line, so the `data-test-deleting-modal` shows and hides instantaneously and the first assert fails.

Replacing `assert.ok(find('[data-test-deleting-modal]'));` with `await waitFor('[data-test-deleting-modal]');` makes it now work with the new helpers.

So the new code becomes:

```js
    let promise = click('[data-test-confirm-delete-modal-button]');

    await waitFor('[data-test-deleting-modal]');

    deferred.resolve();

    await promise;

    assert.notOk(find('[data-test-deleting-modal]'));
```

My question is, is this an unintended bug, or a known think you have to do to use the new helpers?

**Comment by Turbo87** [comment]
> a known think you have to do to use the new helpers

☝️ 

you were relying on the fact that `click()` used to cause a synchronous render, but that is actually not a real guarantee. the helpers in this repo will always run async to enforce that these things are tested correctly.

**Comment by rwjblue** [comment]
This is not a bug at all. `click` should _not_ have synchronous behaviors, and the versions in this repo have explicitly taken steps to ensure that it isn't.

**Comment by rwjblue** [comment]
tldr; what @Turbo87 said 😝

**Comment by kellyselden** [comment]
Well color me educated!
